modules/cluster/cloud-config.yml

```diff
@@ -1,28 +1,80 @@
 #cloud-config
 package_update: true
 packages:
-  - awslogs
   - aws-cfn-bootstrap
+  - unzip
 write_files:
-  - path: "/etc/awslogs/awscli.template"
+  - path: "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json"
     permissions: "0644"
     owner: "root"
     content: |
-      [plugins]
-      cwlogs = cwlogs
-      [default]
-      region = ${region}
-  - path: "/etc/awslogs/awslogs.template"
-    permissions: "0644"
-    owner: "root"
-    content: |
-      [general]
-      state_file = /var/lib/awslogs/agent-state
+      {
+          "agent": {
+              "region": "${region}",
+              "metrics_collection_interval": 60,
+              "logfile": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log"
+          },
+          "metrics": {
+              "metrics_collected": {
+                  "disk": {
+                      "resources": [
+                          "*"
+                      ],
+                      "measurement": [
+                          "disk_used_percent"
+                      ],
+                      "metrics_collection_interval": 60
+                  },
+                  "swap": {
+                      "measurement": [
+                          "swap_used_percent"
+                      ],
+                      "metrics_collection_interval": 5
+                  },
+                  "mem": {
+                      "measurement": [
+                          "mem_used_percent"
+                      ],
+                      "metrics_collection_interval": 5
+                  }
+              },
+              "append_dimensions": {
+                  "ImageId": "$${aws:ImageId}",
+                  "InstanceId": "$${aws:InstanceId}",
+                  "InstanceType": "$${aws:InstanceType}",
+                  "AutoScalingGroupName": "$${aws:AutoScalingGroupName}"
+              },
+              "aggregation_dimensions": [["InstanceId"], ["AutoScalingGroupName"]]
+          },
+          "logs": {
+              "logs_collected": {
+                  "files": {
+                      "collect_list": [
+                          {
+                              "file_path": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log*",
+                              "log_group_name": "${log_group_name}",
+                              "log_stream_name": "{instance_id}/amazon-cloudwatch-agent",
+                              "timezone": "UTC"
+                          },
+                          {
+                              "file_path": "/var/log/amazon/ssm/amazon-ssm-agent.log*",
```

Comment:
Is this data potentially too sensitive to send to cloudwatch? Do the SSM agent logs provide added value?

Reply:
Great question! I looked at the logs now, and turns out they actually log all the command input and output (and the logs don't look all that great). Might be way too chatty about potentially sensitive stuff; just remove that log file?